client/client_test.go

```diff
@@ -2768,6 +2834,11 @@ func testRotateKeySuccess(t *testing.T, serverManagesSnapshotInit bool,
 			keysToExpectCreated = append(keysToExpectCreated, role)
 		}
 	}
+
+	for _, role := range keysToExpectCreated {
```

Comment:
Actually I'm really really sorry to do this after you've made this change, but apparently I may have just forgotten to do something with this because `requireRotationSuccessful` checks that the new key, so possibly we can skip this check entirely?  And also just not collect the expected to be created keys.  :(  Sorry sorry.